scr/main/java/bittercode/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,193 @@
+                String bName = rs.getString(2);
+                String bAuthor = rs.getString(3);
+                int bPrice = rs.getInt(4);
+                int bQty = rs.getInt(5);
+
+                Book book = new Book(bCode, bName, bAuthor, bPrice, bQty);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+
+        }
+        return books;
+    }
+
+    @Override
+    public String deleteBookById(String bookId) throws StoreException {
+        String response = ResponseCode.FAILURE.name();
+        Connection con = DBUtil.getConnection();
+        try {
+            PreparedStatement ps = con.prepareStatement(deleteBookByIdQuery);
+            ps.setString(1, bookId);
+            int k = ps.executeUpdate();
+            if (k == 1) {
+                response = ResponseCode.SUCCESS.name();
+            }
+        } catch (Exception e) {
+            response += " : " + e.getMessage();
+            e.printStackTrace();
+        }
+        return response;
+    }
+
+    @Override
+    public String addBook(Book book) throws StoreException {
+        String responseCode = ResponseCode.FAILURE.name();
+        Connection con = DBUtil.getConnection();
+        try {
+            PreparedStatement ps = con.prepareStatement(addBookQuery);
+            ps.setString(1, book.getBarcode());
+            ps.setString(2, book.getName());
+            ps.setString(3, book.getAuthor());
+            ps.setDouble(4, book.getPrice());
+            ps.setInt(5, book.getQuantity());
+            int k = ps.executeUpdate();
+            if (k == 1) {
+                responseCode = ResponseCode.SUCCESS.name();
+            }
+        } catch (Exception e) {
+            responseCode += " : " + e.getMessage();
+            e.printStackTrace();
+        }
+        return responseCode;
+    }
+
+    @Override
+    public String updateBookQtyById(String bookId, int quantity) throws StoreException {
+        String responseCode = ResponseCode.FAILURE.name();
+        Connection con = DBUtil.getConnection();
+        try {
+            PreparedStatement ps = con.prepareStatement(updateBookQtyByIdQuery);
+            ps.setInt(1, quantity);
+            ps.setString(2, bookId);
+            ps.executeUpdate();
+            responseCode = ResponseCode.SUCCESS.name();
+        } catch (Exception e) {
+            responseCode += " : " + e.getMessage();
+            e.printStackTrace();
+        }
+        return responseCode;
+    }
+
+    @Override
+    public List<Book> getBooksByCommaSeperatedBookIds(String commaSeperatedBookIds) throws StoreException {
+        List<Book> books = new ArrayList<Book>();
+        Connection con = DBUtil.getConnection();
+        try {
+            String getBooksByCommaSeperatedBookIdsQuery = "SELECT * FROM " + BooksDBConstants.TABLE_BOOK
+                    + " WHERE " +
+                    BooksDBConstants.COLUMN_BARCODE + " IN ( " + commaSeperatedBookIds + " )";
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Potential SQL injection vulnerability</b></div>
<div id="fix">

The SQL query is constructed by concatenating strings, which could potentially lead to SQL injection vulnerabilities. Consider using parameterized queries instead.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -151,5 +151,7 @@
          try {
 -            String getBooksByCommaSeperatedBookIdsQuery = "SELECT * FROM " + BooksDBConstants.TABLE_BOOK
 -                    + " WHERE " +
 -                    BooksDBConstants.COLUMN_BARCODE + " IN ( " + commaSeperatedBookIds + " )";
 -            PreparedStatement ps = con.prepareStatement(getBooksByCommaSeperatedBookIdsQuery);
 +            String getBooksByCommaSeperatedBookIdsQuery = "SELECT * FROM " + BooksDBConstants.TABLE_BOOK
 +                    + " WHERE " + BooksDBConstants.COLUMN_BARCODE + " IN (" +
 +                    String.join(",", Collections.nCopies(commaSeperatedBookIds.split(",").length, "?")) + ")";
 +            PreparedStatement ps = con.prepareStatement(getBooksByCommaSeperatedBookIdsQuery);
 +            String[] bookIds = commaSeperatedBookIds.split(",");
 +            for (int i = 0; i < bookIds.length; i++) {
 +                ps.setString(i + 1, bookIds[i].trim());
 +            }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/3#issuecomment-2434584858>#199df6</a></i></small>
</div>


---
🚩 Problem result:
- [ ] This issue was flagged incorrectly